args/core/scheduling/scheduler.cpp

```diff
@@ -6,13 +6,74 @@ namespace args::core::scheduling
 {
     async::readonly_rw_spinlock Scheduler::m_threadsLock;
     sparse_map<std::thread::id, std::unique_ptr<std::thread>> Scheduler::m_threads;
-    const uint Scheduler::m_maxThreadCount = std::thread::hardware_concurrency() == 0 ? std::numeric_limits<uint>::max() : std::thread::hardware_concurrency();
+    std::queue<std::thread::id> Scheduler::m_unreservedThreads;
+    const uint Scheduler::m_maxThreadCount = (((int)std::thread::hardware_concurrency()) - 2) <= 0 ? 4 : std::thread::hardware_concurrency();
     async::readonly_rw_spinlock Scheduler::m_availabilityLock;
     uint Scheduler::m_availableThreads = m_maxThreadCount - 2; // subtract OS and this_thread.
 
-    Scheduler::Scheduler(events::EventBus* eventBus) : m_eventBus(eventBus)
+    async::readonly_rw_spinlock Scheduler::m_jobQueueLock;
+    std::queue<Scheduler::runnable> Scheduler::m_jobs;
+    sparse_map<std::thread::id, async::readonly_rw_spinlock> Scheduler::m_commandLocks;
+    sparse_map<std::thread::id, std::queue<Scheduler::runnable>> Scheduler::m_commands;
+
+    void Scheduler::threadMain(bool* exit, bool* start, bool low_power)
+    {
+        std::thread::id id = std::this_thread::get_id();
+
+        while (!(*start))
+            ;
```

Comment:
why is this `;` on a new line (also why is this not yielding ?)

Reply:
forgot to add yield here will do so before mergs